crates/engine/tree/src/tree/cached_state.rs

```diff
@@ -520,16 +520,16 @@ pub(crate) struct SavedCache {
 
     /// Metrics for the cached state provider
     metrics: CachedStateMetrics,
+
+    /// A guard to track in-flight usage of this cache.
+    /// The cache is considered available if the strong count is 1.
+    pub(crate) usage_guard: Arc<()>,
```

Comment:
can we make this super private, so we don't have to think about where else this can be accessed?